graylog2-web-interface/src/views/components/queries/QueryTitleEditModal.jsx

```diff
@@ -0,0 +1,88 @@
+// @flow strict
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Modal, Button } from 'components/graylog';
+import Input from 'components/bootstrap/Input';
+
+import type { TitlesMap } from 'views/stores/TitleTypes';
+
+/**
+ * Component that allows the user to update a QueryTab title.
+ * It takes the active query title as an argument on open and will use it as the draft.
+ * The open action is getting called outside by referencing this component.
+ */
+
+type Props = {
+  onTitleChange: (newTitle: string) => Promise<TitlesMap>,
+}
+
+type State = {
+  show: boolean,
+  titleDraft: string,
+}
+
+class QueryTitleEditModal extends React.Component<Props, State> {
+  static propTypes = {
+    onTitleChange: PropTypes.func.isRequired,
+  };
+
+  state = {
+    show: false,
+    titleDraft: '',
+  };
+
+  open = (activeQueryTitle: string) => {
+    this.setState({
+      show: true,
+      titleDraft: activeQueryTitle,
+    });
+  }
+
+  _close = () => {
+    this.setState({
+      show: false,
+      titleDraft: '',
+    });
+  }
+
+  _onDraftSave = () => {
+    const { titleDraft } = this.state;
+    const { onTitleChange } = this.props;
+    onTitleChange(titleDraft);
+    this._close();
+  };
+
+  // eslint-disable-next-line no-undef
+  _onDraftChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({ titleDraft: evt.target.value });
+  };
+
+  render() {
+    const { show, titleDraft } = this.state;
+    return (
+      <Modal show={show} bsSize="large" onHide={this._close}>
+        <Modal.Header closeButton>
+          <Modal.Title>Editing query title</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Input autoFocus
+                 help="The title of the query tab."
+                 id="title"
+                 label="Title"
+                 name="title"
+                 onChange={this._onDraftChange}
+                 required
```

Comment:
It just came trough my mind, that this required does nothing, because we are not in the context of a form. I will implement the manual is empty check together with the inout maxlength change, once we are sure we want this restriction.